Makefile

```
@@ -73,41 +49,66 @@ else
 GOBIN=$(shell go env GOBIN)
 endif
 
+# Setting SHELL to bash allows bash commands to be executed by recipes.
+# This is a requirement for 'setup-envtest.sh' in the test target.
+# Options are set to exit when a recipe line exits non-zero or a piped command fails.
+SHELL             = /usr/bin/env bash -o pipefail
+.SHELLFLAGS       = -ec
+
 # GENERATED all: manager
 all: $(SPECIALRESOURCE)
 
-# Run tests
-test: patch # generate fmt vet manifests-gen
-	go test ./... -coverprofile cover.out
+namespace: patch manifests kustomize
+	$(KUSTOMIZE) build config/namespace | kubectl apply -f -
 
-# Build manager binary
-manager: patch generate fmt vet
+##@ Development
+
+manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
+	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+
+generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+
+fmt: ## Run go fmt against code.
+	go fmt ./...
+
+vet: ## Run go vet against code.
+	go vet --mod=vendor ./...
+
+test: patch manifests generate fmt vet envtest ## Run tests.
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./... -coverprofile cover.out
+
+##@ Build
+
+manager: patch generate fmt vet ## Build manager binary.
 	go build -mod=vendor -o /tmp/bin/manager main.go
 
-# Run against the configured Kubernetes cluster in ~/.kube/config
-run: generate fmt vet manifests-gen
+run: manifests generate fmt vet ## Run against the configured Kubernetes cluster in ~/.kube/config
 	go run -mod=vendor ./main.go
 
-configure:
-	# TODO kustomize cannot set name of namespace according to settings, hack TODO
-	cd config/namespace && sed -i 's/name: .*/name: $(NAMESPACE)/g' namespace.yaml
-	cd config/namespace && $(KUSTOMIZE) edit set namespace $(NAMESPACE)
-	cd config/default && $(KUSTOMIZE) edit set namespace $(NAMESPACE)
-	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMAGE)
+local-image-build: patch helm-lint helm-repo-index generate manifests-gen ## Build container image with the manager.
```

Since manifests-gen was renamed to manifests and vise versa, i think this should be adjusted?
```suggestion
local-image-build: patch helm-lint helm-repo-index generate manifests ## Build container image with the manager.
```

---

`local-image-build` does require the contents of the `manifests` directory, which are generated by the `manifests-gen` target. `manifests-gen` in turn requires `manifests`. I would then leave as it; WDYT?

---

I think your logic makes sense.

The way it used to be:
local-image-build <- manifests-gen 

Now we have:
local-image-build <- manifests-gen (used to be called manifests) <- manifests (used to be called manifests-gen)

Considering the `manifests/` directory is copied into the image in the Dockerfile, it makes sense to me to use the new manifests-gen as a dependency... It is a change but I can't think of a reason it would cause any problems.